Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,283 @@
+        if !isManualScrolling {
+            var selectionIndex = 0
+            var testIndex = 1
+            let viewControllerPosOnParent = view.frame.origin.y
+            let currentPos = (scrollView?.contentOffset.y ?? 0) + scrollOffset - viewControllerPosOnParent
+
+            while (testIndex - selectionIndex == 1) && testIndex < originalViews.count {
+
+                if currentPos > originalViews[testIndex].frame.minY {
+                    selectionIndex = testIndex
+                }
+                testIndex += 1
+            }
+
+            tabBar.select(at: selectionIndex)
+        }
+    }
+
+    @objc func filterButtonPressed() {
+        headerContainer.filterButtonPressed()
+
+        if headerContainer.getFilterButtonState() != .inactive {
+            hideTabBar()
+        } else {
+            showTabBar()
+            headerContainer.setFilterLabelText(to: "")
+        }
+    }
+
+    private func hideTabBar() {
+        UIView.animate(withDuration: 0.45) {
+            let adjustedPosTabBar = (self.headerContainer.frame.height / 2) - (self.tabBar.frame.height / 2)
+            let adjustedPosMenuStack = (self.tabBar.frame.height / 2) + (self.menuItemsView.frame.height / 2)
+            self.headerContainer.setSeparatorViewHidden(to: false)
+            self.tabBar.isUserInteractionEnabled = false
+            self.tabBar.center.y = self.headerContainer.center.y + adjustedPosTabBar
+            self.menuItemsView.center.y = self.tabBar.center.y + adjustedPosMenuStack
+        }
+
+        if headerContainer.getFilterButtonState() == .hightolow {
+            scrollView?.contentSize.height -= tabBar.frame.height // Does not animate when subtracted
+            switchToHighToLowStack()
+        } else if headerContainer.getFilterButtonState() == .lowtohigh {
+            switchToLowToHighStack()
+        }
+    }
+
+    private func showTabBar() {
+        UIView.animate(withDuration: 0.45) {
+            let adjustedPosTabBar = (self.headerContainer.frame.height / 2) + (self.tabBar.frame.height / 2)
+            let adjustedPosMenuStack = (self.tabBar.frame.height / 2) + (self.menuItemsView.frame.height / 2)
+            self.headerContainer.setSeparatorViewHidden(to: true)
+            self.tabBar.isUserInteractionEnabled = true
+            self.tabBar.center.y = self.headerContainer.center.y + adjustedPosTabBar
+            self.menuItemsView.center.y = self.tabBar.center.y + adjustedPosMenuStack
+            self.scrollView?.contentSize.height += self.tabBar.frame.height
+        }
+
+        switchToOriginalStack()
+    }
+
+    private func getCompiledItems() -> [ExpandedMenu.Item] {
+        var allMenus: [ExpandedMenu.Item] = []
+        for controller in viewControllers {
+            allMenus += controller.menu
+        }
+
+        return allMenus
+    }
+
+    private func switchToHighToLowStack() {
+        if !orderedViewsCreated {
+            menuItemsView.createHighToLowStack()
+        }
```

Comment:
Is this supposed to set `orderedViewsCreated = true`? 

I think the more idiomatic way would be to make a method `createOrderedViewsIfNeeded()` and abstract this `if` statement and the one below